daffodil-core/src/main/scala/org/apache/daffodil/dsom/SchemaSetIncludesAndImportsMixins.scala

```diff
@@ -57,13 +57,20 @@ trait SchemaSetIncludesAndImportsMixin { self: SchemaSet =>
   lazy val allSchemaFiles = {
     val fd = fakeXMLSchemaDocument //bootstrap
     val sa = fd.seenAfter
+    val (_, first) = if (sa.value.isEmpty) (null, null) else sa.value.head
     val sfl = sa.value.flatMap {
       case (_, ii) => {
         val sf = ii.iiSchemaFileMaybe // maybe not if we've already seen this file for the same namespace.
+        // Require the first schema file to have a DFDL namespace. Other included or imported schemas can be
+        // standard XSD schemas but emit a warning that the schema is being ignored.
         sf.filter { f =>
-          if (f.isDFDLSchemaFile)
+          if (f.isDFDLSchemaFile) {
             true
-          else {
+          } else if (f eq first) {
+            f.SDEButContinue("Non-DFDL Schema file. Does not have DFDL namespace definition on schema root element.\n" +
```

Comment:
Why `SDEButContinue` instead of `SDE`? Seems like we would want to immediately stop if the main schema isn't a DFDL schema.

Reply:
Updated with the recommended change.